substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/Resources.java

```diff
@@ -165,6 +182,68 @@ public static void registerDirectoryResource(String moduleName, String resourceD
         addEntry(moduleName, resourceDirName, true, content.getBytes(), fromJar);
     }
 
+    @Platforms(Platform.HOSTED_ONLY.class)
+    public static void registerIOException(String resourceName, IOException e) {
+        registerIOException(null, resourceName, e);
+    }
+
+    @Platforms(Platform.HOSTED_ONLY.class)
+    public static void registerIOException(String moduleName, String resourceName, IOException e) {
+        Pair<String, String> key = Pair.create(moduleName, resourceName);
+        var resources = singleton().resources;
```

Comment:
This could all be done in a non-static method, avoiding the multiple calls to `singleton()`